docs/doxygen/developer/style.txt

```diff
@@ -0,0 +1,296 @@
+/**
+@page guide_developer_styleguide Style Guide
+@brief Guide for the coding style used in this SDK.
+
+The goal of this style guide is to encourage a readable and consistent coding style across the entire SDK.
+
+@section guide_developer_styleguide_codingstyle Coding Style
+@brief The coding style used in this SDK.
+
+The coding style aims to produce code that is readable and easy to debug. An example is provided in @ref guide_developer_styleguide_codingstyle_example.
+
+@subsection guide_developer_styleguide_codingstyle_generalguidelines General guidelines
+@brief General guidelines for library style.
+- All new libraries (being developed in the development branch) should only use [ANSI C](https://en.wikipedia.org/wiki/ANSI_C#C89) features and standard library functions. (All libraries in the v4_beta branch used the C99 standard, so any changes to the branch can continue using C99).
+- Libraries should log extensively.
+- Code should be well-commented.
+- Only <code>`/`*</code> and <code>*`/`</code> should be used to start and end comments.
+- All comments end with a period.
+- Use of `goto` is discouraged.
+- Only spaces should be used for indenting. A single indent is 4 spaces. No tab characters should be used.
+- A parenthesis is usually followed by a space (see @ref guide_developer_styleguide_codingstyle_example).
+- Lines of code should be less than 80 characters long, although longer lines are permitted.
+- Local variables should be declared at the top of a block in the narrowest scope possible, at the programmer's discretion.
```

Comment:
Maybe we should mention that reducing complexity could be a valid case for not declaring them in the narrowest scope